src/src/PluginStructs/P036_data_struct.cpp

```diff
@@ -590,15 +605,15 @@ bool P036_data_struct::plugin_write(struct EventStruct *event, const String& str
 
       case p036_subcommands_e::userdef1:
       {
-        userDef1 = parseStringKeepCase(string, 3);
-        userDef1.replace('$', '%'); // Allow system vars to be passed in by using $ instead of %
+        userDef1 = ReplaceTokenDelimiters(parseStringKeepCase(string, 3), true); // Allow system vars and task values to be passed to the
+                                                                                 // user defined header1
         break;
       }
 
       case p036_subcommands_e::userdef2:
       {
-        userDef2 = parseStringKeepCase(string, 3);
-        userDef2.replace('$', '%'); // Allow system vars to be passed in by using $ instead of %
+        userDef2 = ReplaceTokenDelimiters(parseStringKeepCase(string, 3), true); // Allow system vars and task values to be passed to the
```

Comment:
the result of `parseString...` should be assigned to the `userDefN` var, and then that should get the replacements.

Reply:
Done, see last commit